### ISSUE TITLE
Add extraThunkArgument to configureStore options

### ISSUE THREAD

**Issue by jedmao** [reduxjs/redux-starter-kit#180]
Related to #113, I would love the ability to provide an [extra thunk argument](https://github.com/reduxjs/redux-thunk#injecting-a-custom-argument) via an `extraThunkArgument` option to `configureStore`.

```ts
const extraThunkArgument = { foo: 'bar' }
const store = configureStore<RootState, RootActions, typeof extraThunkArgument>({
  reducer: rootReducer,
  extraThunkArgument, 
})
```

The result of the thunk should be of the following type:

```ts
ThunkMiddleware<RootState, RootActions, typeof extraThunkArgument>
```

Basically, the types of `configureStore` should match pretty closely the types of [Redux's StoreCreator](https://github.com/reduxjs/redux/blob/9c9a4d2a1c62c9dbddcbb05488f8bd77d24c81de/index.d.ts#L344-L365). A fully typed example of a call to `createStore` would look like this:

```ts
const store = createStore<
	RootState,
	RootActions,
	{
		dispatch: ThunkDispatch<
			RootState,
			typeof extraThunkArgument,
			RootActions
		>
	},
	{}
>(rootReducer, app.loadState())
```

Furthermore, there should be a good strategy for mocking this store, perhaps with a similar approach to [`@jedmao/redux-mock-store`](https://www.npmjs.com/package/@jedmao/redux-mock-store)?

```ts
export default configureMockStore<
  RootState,
  RootActions,
  ThunkDispatch<RootState, typeof extraThunkArgument, RootActions>
>(middlewares)
```

This, however, would be a separate dev dependency and perhaps a separate GitHub issue as well.

**Comment by markerikson** [comment]
Hey, thanks for filing this.

I'm a bit hesitant to explicitly add an option for this directly to `configureStore` itself, for a couple reasons:

- RSK aims to have the most common usages handled by default
- If you need to customize the default middleware, you can override the `middleware` argument and pass in your own list.  That would include a customized `thunk` instance
- Conversely, it's also possible that people might have excluded the thunk middleware from the store setup, so that option might not apply.

Typings-wise, what happens if you do something like `middleware: [thunk.withExtraArgument(someArg)]` ?  How does the resulting store type come out?

**Comment by markerikson** [comment]
Resolved by #192 .
